src/sonic-yang-mgmt/_sonic_yang_ext.py

```diff
@@ -0,0 +1,616 @@
+# This script is used as extension of sonic_yang class. It has methods of
+# class sonic_yang. A separate file is used to avoid a single large file.
+
+import yang as ly
+import re
+import pprint
+
+from json import dump, load, dumps, loads
+from xmltodict import parse
+from os import listdir, walk, path
+from os.path import isfile, join, splitext
+from glob import glob
+
+# class sonic_yang methods
+
+"""
+load all YANG models, create JSON of yang models
+"""
+def loadYangModel(self):
+
+    try:
+        yangDir = self.yang_dir
+        self.yangFiles = glob(yangDir +"/*.yang")
+        for file in self.yangFiles:
+            if (self.load_schema_module(file) == False):
+                return False
+
+        # keep only modules name in self.yangFiles
+        self.yangFiles = [f.split('/')[-1] for f in self.yangFiles]
+        self.yangFiles = [f.split('.')[0] for f in self.yangFiles]
+        print(self.yangFiles)
+
+        # load json for each yang model
+        self.loadJsonYangModel()
+        # create a map from config DB table to yang container
+        self.createDBTableToModuleMap()
+
+    except Exception as e:
+        print("Yang Models Load failed")
+        raise e
+
+    return True
+
+"""
+load JSON schema format from yang models
+"""
+def loadJsonYangModel(self):
+
+    for f in self.yangFiles:
+        m = self.ctx.get_module(f)
+        if m is not None:
+            xml = m.print_mem(ly.LYD_JSON, ly.LYP_FORMAT)
+            self.yJson.append(parse(xml))
+
```

Comment:
Any error handling in this function?

Reply:
This is part of loading Yang model. So I want to print only one msg, thats why error handing is done only in loadYangModel() [Caller func]. I can add error handing here, without printing any msg, so while debugging we can add it.  
Ideally with config command we should pass all debug option, so that we can print all exceptions.